### PR TITLE
TH2-4347 vulnerabilities fix

### PR DESCRIPTION
**Pull request by Xanclry**

### CODE REVIEW

File: src/main/java/com/exactpro/th2/codec/csv/CodecFactory.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 Exactpro (Exactpro Systems Limited)
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/codec/csv/CodecFactory.java:2]: 2023
```suggestion
 * Copyright 2023 Exactpro (Exactpro Systems Limited)
```

File: src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ * Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java:2]:
```suggestion
 * Copyright 2020-2023 Exactpro (Exactpro Systems Limited)
```

File: src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java [hunk 3]
```diff
@@ -83,50 +74,73 @@ public CsvCodec(MessageRouter<MessageGroupBatch> router, MessageRouter<EventBatc
...
+        Collection<ErrorHolder> errors = new ArrayList<>();
+        for (AnyMessage anyMessage : messageGroup.getMessagesList()) {
+            if (anyMessage.hasMessage()) {
+                groupBuilder.addMessages(anyMessage);
+                continue;
+            }
+            if (!anyMessage.hasRawMessage()) {
+                LOGGER.error("Message should either have a raw or parsed message but has nothing: {}", anyMessage);
+                continue;
+            }
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java:90]: Please add protocol check. Codec should decode / encode message with empty or `csv` (case insensitive) protocol

File: src/main/java/com/exactpro/th2/codec/csv/cfg/CsvCodecConfiguration.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ * Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/codec/csv/cfg/CsvCodecConfiguration.java:2]:
```suggestion
 * Copyright 2020-2023 Exactpro (Exactpro Systems Limited)
```

**Comment by OptimumOpium** [comment]
@Nikita-Smirnov-Exactpro @Xanclry Hi, I don't quite understand why are we doing this PR into the master. As I can see the th2-4347 branch is based on the dev branch. Why don't we merge those changes into dev and then merge dev into master?

**Comment by Nikita-Smirnov-Exactpro** [comment]
> @Nikita-Smirnov-Exactpro @Xanclry Hi, I don't quite understand why are we doing this PR into the master. As I can see the th2-4347 branch is based on the dev branch. Why don't we merge those changes into dev and then merge dev into master?

We would like to push this changes ASAP

**Comment by OptimumOpium** [comment]
I understand. Let's switch the base branch to `dev`. Merge changes into `dev` and then merge `dev` into the `master`.
Changes looks good to me
@Nikita-Smirnov-Exactpro If you don't mind I will change the base branch and submit the approval for merge

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**